cli/internal/cmd/upgradeapply.go

```diff
@@ -239,6 +239,13 @@ func (u *upgradeApplyCmd) upgradeApply(cmd *cobra.Command, upgradeDir string, fl
 		return fmt.Errorf("writing state file: %w", err)
 	}
 
+	// TODO(msanft): Remove this after v2.12.0 is released, as we do not support
```

Comment:
Why don't we just remove the file after succesful migration to the state file?

Reply:
If something goes wrong in the process (god knows what), IMO it's better to have a backup file than just remove it